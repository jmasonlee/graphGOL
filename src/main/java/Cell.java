public class Cell implements Comparable {
    public Integer y;
    public Integer x;

    public Cell(Integer x, Integer y) {
        this.x = x;
        this.y = y;
    }

    @Override
    public int compareTo(Object o) {
        return 0;
    }
}
