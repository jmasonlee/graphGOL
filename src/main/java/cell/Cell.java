package cell;

public class Cell implements Comparable {
  public Integer y;
  public Integer x;

  public Cell(Integer x, Integer y) {
    this.x = x;
    this.y = y;
  }

  @Override
  public int compareTo(Object o) {
    return 0;
  }

  @Override
  public boolean equals(Object o) {

    if (o == this) {
      return true;
    }

    if (!(o instanceof Cell)) {
      return false;
    }

    Cell otherCell = (Cell) o;

    return otherCell.x == this.x && otherCell.y == this.y;
  }

  @Override
  public String toString(){
      return "[" + x + "," + y + "]";
  }
}
