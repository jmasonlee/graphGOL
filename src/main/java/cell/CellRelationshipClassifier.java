package cell;

public class CellRelationshipClassifier {
    public static Relationships classify(Cell cell, Cell cell1) {
        return Relationships.SELF;
    }

    public enum Relationships {
        DISCONNECTED(-1), SELF(0), NEIGHBOUR(1), COPARENT(2);

        public final int maxDistance;

        private Relationships(int maxDistance){
            this.maxDistance = maxDistance;
        }
    }


}
