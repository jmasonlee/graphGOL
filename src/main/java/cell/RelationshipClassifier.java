package cell;

public class RelationshipClassifier {
    public static Relationships classify(Cell cell, Cell cell1) {
        return Relationships.SELF;
    }

    public enum Relationships {
        DISCONNECTED(-1), SELF(0), NEIGHBOUR(1), COPARENT(2);

        public final int maxDistance;

        private Relationships(int maxDistance){
            this.maxDistance = maxDistance;
        }
    }


}
