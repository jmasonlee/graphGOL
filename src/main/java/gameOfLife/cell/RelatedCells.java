package gameOfLife.cell;

import java.util.List;
import java.util.Map;

public class RelatedCells {
  private final Map<Relationships, List<Cell>> relationships;

  public RelatedCells(Map<Relationships, List<Cell>> relationships) {
    this.relationships = relationships;
  }

  public Map<Relationships, List<Cell>> getRelationships() {
    return relationships;
  }

  @Override
  public String toString() {
    return relationships.toString();
  }

  public List<Cell> getCellsOfRelationshipType(Relationships relationshipType) {
    return relationships.get(relationshipType);
  }
}
