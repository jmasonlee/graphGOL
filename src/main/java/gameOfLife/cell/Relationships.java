package gameOfLife.cell;

public enum Relationships {
  DISCONNECTED(-1),
  SELF(0),
  NEIGHBOUR(Math.sqrt(2)),
  COPARENT(Math.sqrt(8));

  public final double maxDistance;

  Relationships(double maxDistance) {
    this.maxDistance = maxDistance;
  }
}
