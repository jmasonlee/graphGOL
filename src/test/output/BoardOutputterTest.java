package output;

import cell.Cell;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.Assert.assertEquals;

public class BoardOutputterTest {
  @Test
  public void testBoardIsGeneratedWithCellsInCorrectLocation() {
    Integer[] coordinateNumbers = new Integer[]{0, 1, 2, 3, 4};

    CombinationApprovals.verifyAllCombinations(
        this::createBoardWithCellInLocation, coordinateNumbers, coordinateNumbers);
  }

  @Test
  public void testBoardIsGeneratedWithCellsInCorrectLocationIfCellIsInNegativeSpace() {
    String expectedBoardOutput = "X    \n     \n     \n     \n     ";
    assertEquals(expectedBoardOutput, BoardOutputter.createBoardOutput(Arrays.asList(new Cell(-2, -1))));
  }

  private String createBoardWithCellInLocation(Integer x, Integer y) {
    return "\n"+BoardOutputter.createBoardOutput(Arrays.asList(new Cell(0,0), new Cell(x, y)));
  }
}
