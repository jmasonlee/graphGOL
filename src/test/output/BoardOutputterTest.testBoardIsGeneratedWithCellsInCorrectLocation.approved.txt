[0, 0] => 
X    
     
     
     
      
[0, 1] => 
X    
X    
     
     
      
[0, 2] => 
X    
     
X    
     
      
[0, 3] => 
X    
     
     
X    
      
[0, 4] => 
X    
     
     
     
X     
[1, 0] => 
XX   
     
     
     
      
[1, 1] => 
 X   
X    
     
     
      
[1, 2] => 
 X   
     
X    
     
      
[1, 3] => 
 X   
     
     
X    
      
[1, 4] => 
 X   
     
     
     
X     
[2, 0] => 
X X  
     
     
     
      
[2, 1] => 
  X  
X    
     
     
      
[2, 2] => 
  X  
     
X    
     
      
[2, 3] => 
  X  
     
     
X    
      
[2, 4] => 
  X  
     
     
     
X     
[3, 0] => 
X  X 
     
     
     
      
[3, 1] => 
   X 
X    
     
     
      
[3, 2] => 
   X 
     
X    
     
      
[3, 3] => 
   X 
     
     
X    
      
[3, 4] => 
   X 
     
     
     
X     
[4, 0] => 
X   X
     
     
     
      
[4, 1] => 
    X
X    
     
     
      
[4, 2] => 
    X
     
X    
     
      
[4, 3] => 
    X
     
     
X    
      
[4, 4] => 
    X
     
     
     
X     
